manual/manual/tutorials/coreexamples.etex

```diff
@@ -707,7 +707,18 @@ let str : _ format =
 Printf.printf str 3 4.5 "string value";;
 \end{caml_example}
 "fprintf" function is like "printf" except that it takes an output channel as
-the first argument. Here is how to rewrite the pretty-printer using "fprintf":
+the first argument.
```

Comment:
I think we should explain here what is going on in the preliminary example:

- Explain what are you doing defining the `output_int` printer

-  then mention hat this approach works for more complex types, for example options, and thst at in this case you also need to provide a printer for the type that they wrap,

- then explain that in this way you are constructing composable printers that can be used with the `%a` format specifier and conclude the example.

This will greatly clarify for the reader both the introductory examples that you just added, and the more complex `print_expr` presented afterwards

Reply:
I agree with @mseri that the next example lacks a short introduction.
For instance,
> The `%a` specifier can be useful to define custom printer (for custom types). For instance, a custom printer for ints:

Then gives an example of using `pp_int` with `%a`, before introducing `pp_option` as a combinator that transforms a printer for `'a` into a printer for `'a option`.

Reply:
@Octachron Sorry, but I didn't understand what kind of example is needed. Should I somehow wrap int to use it with %a?

Reply:
The example with `pp_int`? I was thinking of something like:
```OCaml
\begin{caml_example}{toplevel}
let pp_int ppf n = Printf.fprintf ppf "%d" n;;
Printf.printf "Using a custom printer for integer %a\n" pp_int 12
\end{caml_example}{toplevel}
```
to show that `pp_int` is a composable printer by itself.

Reply:
I agree. I would also split the example in its consituents, somewhat like:
```latex
[...]the first argument. This is useful to construct composable printers that can be used with the "%a" specifier.
For example, we can create a printing template that converts an integer argument to signed decimal as follows.
\begin{caml_example}{toplevel}
let pp_int ppf n = Printf.fprintf ppf "%d" n;;
Printf.printf "Using a custom printer for integer %a\n" pp_int 12
\end{caml_example}{toplevel}
As you can see, the printing template is a composable printer by itself. We can push this example further, and define a printer for optional types.
\begin{caml_example}{toplevel}
let pp_option printer ppf = function
  | None -> Printf.fprintf ppf "None"
  | Some v -> Printf.fprintf ppf "Some(%a)" printer v
+ couple of examples of use if you want
\end{caml_example}{toplevel}
Here you see that we need to also provide a printer as argument, indeed "pp_option" is a function that processes the argument passed to it and depending on its value prints "None"  or usest the provided printer.
[...]
```
Guiding the reader through the example, instead of showing the example and then explaining it as a whole.

Reply:
Don't get me wrong, this is already a very good addition to the manual, but I think that a small restructuring can make it even better

Reply:
Thank you very much. I added explanations and restructured them.